File: app/src/main/kotlin/com/foodshare/ui/design/tokens/LiquidGlassSpacing.kt
package com.foodshare.ui.design.tokens

import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * Liquid Glass Spacing & Layout System v26
 *
 * Based on 8pt grid system
 * Ported from iOS: LiquidGlassSpacing.swift
 */
object Spacing {
    // MARK: - Base Unit (8dp grid)
    val unit: Dp = 8.dp

    // MARK: - Semantic Spacing
    val xxxs: Dp = 4.dp   // unit * 0.5
    val xxs: Dp = 8.dp    // unit
    val xs: Dp = 12.dp    // unit * 1.5
    val sm: Dp = 16.dp    // unit * 2
    val md: Dp = 24.dp    // unit * 3
    val lg: Dp = 32.dp    // unit * 4
    val xl: Dp = 40.dp    // unit * 5
    val xxl: Dp = 48.dp   // unit * 6
    val xxxl: Dp = 64.dp  // unit * 8

    // MARK: - Component-specific spacing
    val buttonPaddingHorizontal: Dp = md
    val buttonPaddingVertical: Dp = sm
    val cardPadding: Dp = md
    val listItemPadding: Dp = sm
    val screenPadding: Dp = md
    val sectionSpacing: Dp = lg
}

/**
 * Corner Radius tokens for consistent rounding
 */
object CornerRadius {
    /** 4dp - extra small (badges, pills) */
    val xs: Dp = 4.dp

    /** 8dp - small (chips, small cards) */
    val small: Dp = 8.dp

    /** 12dp - medium (cards, inputs) */
    val medium: Dp = 12.dp

    /** 16dp - large (modals, sheets) */
    val large: Dp = 16.dp

    /** 20dp - extra large */
    val xl: Dp = 20.dp

    /** 24dp - extra extra large (full-screen overlays) */
    val xxl: Dp = 24.dp

    /** 28dp - for large buttons */
    val button: Dp = 28.dp

    /** 9999dp - fully rounded (pills, avatars) */
    val full: Dp = 9999.dp
}

/**
 * Shadow elevation tokens
 */
object Elevation {
    val none: Dp = 0.dp
    val xs: Dp = 1.dp
    val sm: Dp = 2.dp
    val md: Dp = 4.dp
    val lg: Dp = 8.dp
    val xl: Dp = 16.dp
    val xxl: Dp = 24.dp
}

/**
 * Component size tokens
 */
object ComponentSize {
    // Buttons
    val buttonHeightSmall: Dp = 36.dp
    val buttonHeightMedium: Dp = 48.dp
    val buttonHeightLarge: Dp = 56.dp
    val buttonHeightXLarge: Dp = 58.dp

    // Icons
    val iconXs: Dp = 16.dp
    val iconSm: Dp = 20.dp
    val iconMd: Dp = 24.dp
    val iconLg: Dp = 32.dp
    val iconXl: Dp = 48.dp

    // Avatars
    val avatarSm: Dp = 32.dp
    val avatarMd: Dp = 48.dp
    val avatarLg: Dp = 64.dp
    val avatarXl: Dp = 100.dp

    // Cards
    val cardImageHeightCompact: Dp = 120.dp
    val cardImageHeightStandard: Dp = 180.dp
    val cardImageHeightFeatured: Dp = 220.dp

    // Inputs
    val inputHeight: Dp = 56.dp
    val inputHeightLarge: Dp = 64.dp

    // Bottom navigation
    val bottomNavHeight: Dp = 80.dp
    val tabBarHeight: Dp = 56.dp
}

File: Sources/FoodShare/Core/Design/Tokens/LiquidGlassSpacing.swift
//
//  LiquidGlassSpacing.swift
//  Foodshare
//
//  Liquid Glass Spacing & Layout System v26
//

import SwiftUI
import FoodShareDesignSystem

enum Spacing {
    // MARK: - Base Unit (8pt grid)
    static let unit: CGFloat = 8

    // MARK: - Semantic Spacing
    static let xxxs: CGFloat = unit * 0.5 // 4
    static let xxs: CGFloat = unit // 8
    static let xs: CGFloat = unit * 1.5 // 12
    static let sm: CGFloat = unit * 2 // 16
    static let md: CGFloat = unit * 3 // 24
    static let lg: CGFloat = unit * 4 // 32
    static let xl: CGFloat = unit * 5 // 40
    static let xxl: CGFloat = unit * 6 // 48
    static let xxxl: CGFloat = unit * 8 // 64

    // MARK: - Corner Radius
    static let radiusXS: CGFloat = 4
    static let radiusSM: CGFloat = 8
    static let radiusMD: CGFloat = 12
    static let radiusLG: CGFloat = 16
    static let radiusXL: CGFloat = 24
    static let radiusFull: CGFloat = 9999

    // MARK: - Shadows
    static let shadowSM: CGFloat = 2
    static let shadowMD: CGFloat = 4
    static let shadowLG: CGFloat = 8
    static let shadowXL: CGFloat = 16
}

// MARK: - LiquidGlassSpacing Alias
typealias LiquidGlassSpacing = Spacing

// MARK: - Corner Radius

enum CornerRadius {
    /// 8pt - small components (badges, chips)
    static let small: CGFloat = 8
    /// 12pt - medium components (cards, inputs)
    static let medium: CGFloat = 12
    /// 16pt - large components (modals, sheets)
    static let large: CGFloat = 16
    /// 24pt - extra large components (full-screen overlays)
    static let xl: CGFloat = 24
    /// 28pt - extra extra large (featured cards)
    static let xxl: CGFloat = 28
    /// Fully rounded (pills, avatars)
    static let full: CGFloat = 9999

    // MARK: - Deprecated Aliases

    @available(*, deprecated, renamed: "large")
    static let lg: CGFloat = 16
    @available(*, deprecated, renamed: "xl")
    static let extraLarge: CGFloat = 24
}
